src/pythia/src/jobs/publisher.rs

```diff
@@ -0,0 +1,142 @@
+use std::{time::Duration, vec};
+
+use anyhow::{Context, Result};
+use ic_cdk::export::candid::Nat;
+use ic_cdk_timers::set_timer;
+use ic_dl_utils::retry_until_success;
+
+use futures::future::join_all;
+
+use super::withdraw;
+use crate::{
+    clone_with_state, log,
+    types::{
+        balance::Balances,
+        errors::PythiaError,
+        subscription::{Subscription, Subscriptions},
+        timer::Timer,
+    },
+    utils::{
+        abi, address, canister,
+        multicall::{multicall, Call},
+        nat, web3,
+    },
+};
+
+pub fn execute() {
+    ic_cdk::spawn(async {
+        if let Err(e) = _execute().await {
+            log!("error while executing publisher job: {e:?}");
+        }
+    })
+}
+
+async fn _execute() -> Result<()> {
+    log!("[PUBLISHER] publisher job started");
+    Timer::activate().context(PythiaError::UnableToActivateTimer)?;
+
+    Subscriptions::stop_insufficients()
+        .context(PythiaError::UnableToStopInsufficientSubscriptions)?;
+
+    let mut futures = vec![];
+    let (publishable_subs, is_active) = Subscriptions::get_publishable();
+    publishable_subs.iter().for_each(|(chain_id, subs)| {
```

Comment:
This could be done a little bit simpler:

```rust
let futures = publishable_subs.into_iter()
                    .filter(|(_, subs)| subs.is_empty())
                    .map(|(chain_id, subs)| publish_on_chain(chain_id, subs))
                    .collect::<Vec<_>>();
```